frontend/src/components/LoginForm.tsx

```diff
@@ -77,4 +78,70 @@ const LoginForm = () => {
   );
 };
 
-export default LoginForm;
+export const SessionLoginForm = () => {
+  const { sessionId, accessCode } = useParams<{ sessionId: string; accessCode: string }>();
+  const { user, updateUserId, updateSessionId, updateAccessCode, isLecturer, updateUser } = useUser();
+
+  const history = useHistory();
+
+  const handleSubmit = async (event: any) => {
+    event.preventDefault();
+
+    try {
+      const response = await loginService.login({
+        email: event.target.email.value,
+        password: event.target.password.value,
+      });
+
+      const decodedUser: User = jwt(response.token);
+      updateUser(decodedUser);
+      sessionStorage.setItem('jwt', response.token);
+      sessionStorage.setItem('email', decodedUser.email);
+      sessionStorage.setItem('role', decodedUser.role);
+      sessionStorage.setItem('firstName', decodedUser.firstName);
+      sessionStorage.setItem('lastName', decodedUser.lastName);
+      sessionStorage.setItem('sessionId', sessionId);
+      sessionStorage.setItem('accessCode', accessCode);
```

Comment:
@szarbartosz co powiedziałbyś na `sessionStorage.setItem('user', JSON.stringify(decodedUser));` przy zapisie oraz
`JSON.parse(sessionStorage.getItem('user'));` przy odczycie. liczba odwołań do session storage'a zmaleje wtedy z 7 do 2
Zostawiam jako sugestię

Reply:
@kpodsiad bardzo dobra sugestia, jak najbardziej się z nią zgadzam - poprawię to przy okazji naszej pracy z quizem